### DISCUSSION TITLE
ACP-23: P-Chain Native Transfers

### DISCUSSION THREAD

**Discussion by avalanche-foundation-admin** [avalanche-foundation/ACPs#37] [Discussions]
Support native transfers on P-chain. This enables users to transfer P-chain assets without leaving the P-chain or using a transaction type that's not meant for native transfers.

https://github.com/avalanche-foundation/ACPs/blob/main/ACPs/23-p-chain-native-transfers/README.md

**Comment by kitpub** [comment]
Glad to see the mention of possible DoS attacks in the future due to the low fee, as this was my only concern. Dynamic transaction fees in the future make sense for this very important chain. Otherwise, looks great!

**Comment by neirenoir** [comment]
At the risk of sounding like a Negative Nancy, wouldn't this be kind of feature creep-ish? This is what the X-chain is for. If the P-chain supports cheap and quick transactions as well, why would I ever want to leave the P-chain to use the X-chain?

**Reply by dhrubabasu** [reply]
If you are only interested in staking, you shouldn't have to leave the P-Chain. The demand for native transfers on P-Chain is there as validators are using other transactions not meant for native transfers to replicate its functionality.

The feature-creep comment is noted but I believe that's not a big concern for this particular change. The X-Chain is used for more than just cheap and quick transactions, see here for more details: https://docs.avax.network/learn/avalanche/avalanche-platform#x-chain

**Comment by d10r** [comment]
With the P-Chain maybe becoming the only chain validated by all subnet validators, it could become attractive for folks like inscriptors.
Since resource isolation is a central motivator of the subnet architecture, protecting validator management txs and AWM txs against disruptions caused by P-Chain overloading should have utmost priority.

I think enabling P-Chain transfers makes it more vulnerable to the kind of uses which happen to effectively result in DoS as a side effect.
I'm not strictly against enabling P-Chain transfers (in the context of #14) , but not comfortable about doing this without also adjusting the fee dynamics. Postponing that to later likely means that it will happen only after there was an incident. And in case there is such an incident, the P-Chain may forever have to carry the bloat added in the meantime, or alternatively need an ugly hardfork for removing it.

A possibly simple way to reconcile and enable the feature in the short term could be to allow P-Chain transfers, but dedicate only a rather limited amount of blockspace to that, such that even without a more sophisticated fee mechanism there's no risk for it to disrupt the more important P-Chain activity, nor to bloat it much.
